tket/src/WeightSubgrMono/GraphTheoretic/CloseNeighboursCalculator.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019-2022 Cambridge Quantum Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "WeightSubgrMono/GraphTheoretic/CloseNeighboursCalculator.hpp"
+
+#include <algorithm>
+
+#include "WeightSubgrMono/GraphTheoretic/NeighboursData.hpp"
+
+namespace tket {
+namespace WeightedSubgraphMonomorphism {
+
+void CloseNeighboursCalculator::operator()(
+    unsigned num_levels, CloseVertices& close_vertices,
+    const NeighboursData& ndata, VertexWSM v) {
+  close_vertices.resize(num_levels);
+  if (close_vertices.empty()) {
+    return;
+  }
+  close_vertices[0] = ndata.get_neighbours_expensive(v);
```

Comment:
It's a bad choice of names maybe, but get_neighbours_expensive is already doing exactly what you say, it's just a wrapper. It's "expensive" because it is O(N), reconstructed anew each time, whereas neighbours_and_weights is O(1) - it returns a reference to stored const data.

Reply:
Nice, thanks for the explanation.